src/lib/fft.jl

```diff
@@ -0,0 +1,57 @@
+
+using FFTW
+using FillArrays
+# FFTW functions do not work with FillArrays, which are needed
+# for some functionality of Zygote. To make it work with FillArrays
+# as well, overload the relevant functions
+FFTW.fft(x::Fill) = FFTW.fft(collect(x))
+FFTW.ifft(x::Fill) = FFTW.ifft(collect(x))
+FFTW.fft(x::Fill, dims) = FFTW.fft(collect(x), dims)
+FFTW.ifft(x::Fill, dims) = FFTW.ifft(collect(x), dims)
+
+
+# the adjoint jacobian of an FFT with respect to its input is the reverse FFT of the
+# gradient of its inputs, but with different normalization factor
+@adjoint function FFTW.fft(xs)
+    return FFTW.fft(xs), function(Δ)
+        N = length(xs)
+        return (N * FFTW.ifft(Δ),)
+    end
+end
+
+@adjoint function FFTW.ifft(xs)
+    return FFTW.ifft(xs), function(Δ)
+        N = length(xs)
+        return (1/N* FFTW.fft(Δ),)
+    end
+end
+
+@adjoint function FFTW.fft(xs, dims)
+    # up to now only works when dims is a single integer
```

Comment:
This comment isn't clear to me; you seem to be handling multiple dimensions below. If only single integers work, best to add a type restriction here.

Reply:
this comment was from an old version and I forgot to remove it, sorry for the confusion. It indeed works also with multiple dimensions. I removed the comment now